File: app/src/main/java/cn/mijack/imagedrive/entity/FirebaseImage.java
package cn.mijack.imagedrive.entity;

import android.os.Build;
import android.os.Parcel;
import android.os.Parcelable;

import java.util.HashMap;
import java.util.Map;

/**
 * @author Mr.Yuan
 * @date 2017/5/14
 */
public class FirebaseImage implements Parcelable {
    private String miniType;
    private long dateTaken;
    private String downloadUrl;
    private String fsUrl;
    private String deviceId;
    private int width;
    private int height;
    private String localPath;
    private String name;
    private String device;
    private long uploadTime;

    public FirebaseImage() {
    }

    public FirebaseImage(Image image, String downloadUrl, String fsUrl) {
        this.downloadUrl = downloadUrl;
        this.fsUrl = fsUrl;
        this.dateTaken = image.getDateTaken();
        this.width = image.getWidth();
        this.height = image.getHeight();
        this.localPath = image.getPath();
        this.name = image.getName();
        this.miniType = image.getMiniType();
        this.device = Build.DEVICE;
        this.deviceId = Build.ID;
    }

    protected FirebaseImage(Parcel in) {
        miniType = in.readString();
        dateTaken = in.readLong();
        downloadUrl = in.readString();
        fsUrl = in.readString();
        deviceId = in.readString();
        width = in.readInt();
        height = in.readInt();
        localPath = in.readString();
        name = in.readString();
        device = in.readString();
        uploadTime = in.readLong();
    }

    public static final Creator<FirebaseImage> CREATOR = new Creator<FirebaseImage>() {
        @Override
        public FirebaseImage createFromParcel(Parcel in) {
            return new FirebaseImage(in);
        }

        @Override
        public FirebaseImage[] newArray(int size) {
            return new FirebaseImage[size];
        }
    };

    public String getDownloadUrl() {
        return downloadUrl;
    }

    public void setDownloadUrl(String downloadUrl) {
        this.downloadUrl = downloadUrl;
    }

    public String getFsUrl() {
        return fsUrl;
    }

    public void setFsUrl(String fsUrl) {
        this.fsUrl = fsUrl;
    }

    public String getDeviceId() {
        return deviceId;
    }

    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public String getLocalPath() {
        return localPath;
    }

    public void setLocalPath(String localPath) {
        this.localPath = localPath;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDevice() {
        return device;
    }

    public void setDevice(String device) {
        this.device = device;
    }

    public long getDateTaken() {
        return dateTaken;
    }

    public void setDateTaken(long dateTaken) {
        this.dateTaken = dateTaken;
    }

    public long getUploadTime() {
        return uploadTime;
    }

    public void setUploadTime(long uploadTime) {
        this.uploadTime = uploadTime;
    }

    public String getMiniType() {
        return miniType;
    }

    public void setMiniType(String miniType) {
        this.miniType = miniType;
    }

    public Map<String, Object> toMap() {
        uploadTime = System.currentTimeMillis();
        Map<String, Object> map = new HashMap<>();
        map.put("uploadTime", this.uploadTime);
        map.put("downloadUrl", this.downloadUrl);
        map.put("fsUrl", this.fsUrl);
        map.put("width", this.width);
        map.put("height", this.height);
        map.put("localPath", this.localPath);
        map.put("name", this.name);
        map.put("device", this.device);
        map.put("deviceId", this.deviceId);
        map.put("miniType", this.miniType);
        map.put("dateTaken", this.dateTaken);
        return map;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(miniType);
        dest.writeLong(dateTaken);
        dest.writeString(downloadUrl);
        dest.writeString(fsUrl);
        dest.writeString(deviceId);
        dest.writeInt(width);
        dest.writeInt(height);
        dest.writeString(localPath);
        dest.writeString(name);
        dest.writeString(device);
        dest.writeLong(uploadTime);
    }
}

File: app/src/main/java/cn/mijack/imagedrive/entity/FirebaseImage.kt
package cn.mijack.imagedrive.entity

import android.os.Build
import android.os.Parcel
import android.os.Parcelable
import java.util.*

/**
 * @author admin
 * @date 2017/8/26
 */
class FirebaseImage(var miniType: String, var dateTaken: Long,
                    var downloadUrl: String, var fsUrl: String,
                    var deviceId: String, var width: Int, var height: Int,
                    var localPath: String, var name: String, var device: String, var uploadTime: Long) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readLong(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readInt(),
            parcel.readInt(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readLong())

    constructor(image: Image, downloadUrl: String, fsUrl: String) :
            this(image.miniType!!, image.dateTaken, downloadUrl, fsUrl, Build.ID,
                    image.width, image.height, image.path, image.name!!,
                    Build.DEVICE, null!!)

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(miniType)
        parcel.writeLong(dateTaken)
        parcel.writeString(downloadUrl)
        parcel.writeString(fsUrl)
        parcel.writeString(deviceId)
        parcel.writeInt(width)
        parcel.writeInt(height)
        parcel.writeString(localPath)
        parcel.writeString(name)
        parcel.writeString(device)
        parcel.writeLong(uploadTime)
    }

    override fun describeContents(): Int {
        return 0
    }

    fun toMap(): Map<String, Any> {
        uploadTime = System.currentTimeMillis()
        var map = HashMap<String, Any>()
        map.put("uploadTime", this.uploadTime)
        map.put("downloadUrl", this.downloadUrl)
        map.put("fsUrl", this.fsUrl)
        map.put("width", this.width)
        map.put("height", this.height)
        map.put("localPath", this.localPath)
        map.put("name", this.name)
        map.put("device", this.device)
        map.put("deviceId", this.deviceId)
        map.put("miniType", this.miniType)
        map.put("dateTaken", this.dateTaken)
        return map
    }

    companion object CREATOR : Parcelable.Creator<FirebaseImage> {
        override fun createFromParcel(parcel: Parcel): FirebaseImage {
            return FirebaseImage(parcel)
        }

        override fun newArray(size: Int): Array<FirebaseImage?> {
            return arrayOfNulls(size)
        }
    }
}